tests/test_tags.py

```diff
@@ -1,75 +1,99 @@
+    builder = wheel_builder_test_factory(monkeypatch, {
+        'platlib': [f'extension{SUFFIX}'],
+    })
+    assert str(builder.tag) == f'{INTERPRETER}-{ABI}-{PLATFORM}'
+
+
+@pytest.mark.skipif(not ABI3SUFFIX, reason='Stable ABI not supported by Python interpreter')
+def test_tag_stable_abi(monkeypatch):
+    builder = wheel_builder_test_factory(monkeypatch, {
+        'platlib': [f'extension{ABI3SUFFIX}'],
+    })
+    assert str(builder.tag) == f'{INTERPRETER}-abi3-{PLATFORM}'
+
+
+@pytest.mark.skipif(not ABI3SUFFIX, reason='Stable ABI not supported by Python interpreter')
+def test_tag_mixed_abi(monkeypatch):
+    builder = wheel_builder_test_factory(monkeypatch, {
+        'platlib': [f'extension{ABI3SUFFIX}', f'another{SUFFIX}'],
+    })
+    assert str(builder.tag) == f'{INTERPRETER}-{ABI}-{PLATFORM}'
+
+
+@pytest.mark.skipif(platform.system() != 'Darwin', reason='macOS specific test')
+def test_tag_macos_build_target(monkeypatch):
+    monkeypatch.setenv('MACOS_BUILD_TARGET', '12.0')
+    builder = wheel_builder_test_factory(monkeypatch, {
+        'platlib': [f'extension{SUFFIX}'],
+    })
+    assert builder.tag.platform == re.sub(r'\d+\.\d+', '12.0', PLATFORM)
 
-INTERPRETER_VERSION = f'{sys.version_info[0]}{sys.version_info[1]}'
-
-
-@pytest.mark.parametrize(
-    ('value', 'number', 'abi', 'python'),
-    [
-        ('abi3', 3, 'abi3', None),
-        ('abi4', 4, 'abi4', None),
-    ]
-)
-def test_stable_abi_tag(value, number, abi, python):
-    tag = mesonpy._tags.StableABITag(value)
-    assert str(tag) == value
-    assert tag.abi_number == number
-    assert tag.abi == abi
-    assert tag.python == python
-    assert tag == mesonpy._tags.StableABITag(value)
-
-
-def test_stable_abi_tag_invalid():
-    with pytest.raises(ValueError, match=re.escape(
-        r'Invalid PEP 3149 stable ABI tag, expecting pattern `^abi(?P<abi_number>[0-9]+)$`'
-    )):
-        mesonpy._tags.StableABITag('invalid')
-
-
-@pytest.mark.parametrize(
-    ('value', 'implementation', 'version', 'additional', 'abi', 'python'),
-    [
-        ('cpython-37-x86_64-linux-gnu', 'cpython', '37', ('x86_64', 'linux', 'gnu'), 'cp37', 'cp37'),
-        ('cpython-310-x86_64-linux-gnu', 'cpython', '310', ('x86_64', 'linux', 'gnu'), 'cp310', 'cp310'),
-        ('cpython-310', 'cpython', '310', (), 'cp310', 'cp310'),
-        ('cp311-win_amd64', 'cpython', '311', ('win_amd64', ), 'cp311', 'cp311'),
-        ('cpython-311-win_amd64', 'cpython', '311', ('win_amd64', ), 'cp311', 'cp311'),
-        ('cpython-310-special', 'cpython', '310', ('special',), 'cp310', 'cp310'),
-        ('cpython-310-x86_64-linux-gnu', 'cpython', '310', ('x86_64', 'linux', 'gnu'), 'cp310', 'cp310'),
-        ('pypy39-pp73-x86_64-linux-gnu', 'pypy39', 'pp73', ('x86_64', 'linux', 'gnu'), 'pypy39_pp73', 'pp39'),
-        ('pypy39-pp73-win_amd64', 'pypy39', 'pp73', ('win_amd64', ), 'pypy39_pp73', 'pp39'),
-        ('pypy38-pp73-darwin', 'pypy38', 'pp73', ('darwin', ), 'pypy38_pp73', 'pp38'),
```

Comment:
These should be kept, right? We had a lot of trouble with wheel tags, so I'd like to make sure there are no regressions - that would make packages immediately uninstallable from source.

Reply:
This was testing the conversion from a extension module filename suffix to a wheel tag. This approach cannot work as the filename suffix does not carry enough information (and strictly speaking it is not mandatory for it to carry any). This patch set moves away from that approach. Thus this test is not necessary anymore and the code that is being tested does not exist anymore.